numpy/core/src/multiarray/methods.c

```diff
@@ -358,15 +358,22 @@ NPY_NO_EXPORT PyObject *
 PyArray_GetField(PyArrayObject *self, PyArray_Descr *typed, int offset)
 {
     PyObject *ret = NULL;
+    PyObject *_numpy_internal, *safe;
```

Comment:
This is usually done using a static variable initialized on first call. For instance
```
    static PyObject *matmul = NULL;

    if (matmul == NULL) {
        PyObject *module, *dict;

        module = PyImport_ImportModule("numpy.core.multiarray");
        dict = PyModule_GetDict(module);
        matmul = PyDict_GetItemString(dict, "matmul");
        Py_INCREF(matmul);
        Py_DECREF(module);
    }
```